### PR TITLE
port detection by tracker

### PR DESCRIPTION
**Pull request by yukke42**
## Description

.auto porting

- DynamicObjectWithFeatureArray -> DetectedObjects, DetectedObjectsWithFeatrue and TrackedObjects

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [pull request guidelines][pull-request-guidelines]
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: perception/object_recognition/detection/detection_by_tracker/src/detection_by_tracker_core.cpp [hunk 1]
```diff
@@ -342,33 +372,33 @@ float DetectionByTracker::optimizeUnderSegmentedObject(
   }
 
   // build output
-  highest_iou_object.object.semantic = target_object.semantic;
-  highest_iou_object.object.semantic.confidence = 0.1f;
+  highest_iou_object.object.classification = target_object.classification;
+  highest_iou_object.object.existence_probability = 0.1f;
```
Comment (yukkysaito) [inline perception/object_recognition/detection/detection_by_tracker/src/detection_by_tracker_core.cpp:376]: //TODO It is necessary to consider appropriate values in the future.

みたいな文言お願いできますか？

Comment (yukke42) [reply perception/object_recognition/detection/detection_by_tracker/src/detection_by_tracker_core.cpp:376]: Is this message neccesary for other hard-coded values? s.t. recall_min_threshold

Comment (yukkysaito) [reply perception/object_recognition/detection/detection_by_tracker/src/detection_by_tracker_core.cpp:376]: 一旦ここだけで大丈夫です！

Comment (yukke42) [reply perception/object_recognition/detection/detection_by_tracker/src/detection_by_tracker_core.cpp:377]: add at [4e9b7a0](https://github.com/tier4/AutowareArchitectureProposal.iv/pull/589/commits/4e9b7a0222104b8cff0d8c400156d704df1fda58)

**Review by yukkysaito — APPROVED** [review]

**Pull request merged**